sherpa/astro/ui/tests/test_astro_ui_io.py

```diff
@@ -22,90 +22,83 @@
-class test_load_pha3_gzip(SherpaTestCase):
-    """Handle a .gz FITS PHA Type 3 file"""
-
-    longMessage = True
+FILE_NAME = 'acisf01575_001N001_r0085_pha3.fits'
 
-    def setUp(self):
-        # hide warning messages from file I/O
-        self._old_logger_level = logger.level
-        logger.setLevel(logging.ERROR)
 
-        ui.clean()
+def validate_pha(idval):
+    """Check that the PHA dataset in id=idval is
+    as expected.
+    """
 
-        self.head = self.make_path('acisf01575_001N001_r0085')
+    assert ui.list_data_ids() == [idval]
 
-    def tearDown(self):
-        logger.setLevel(self._old_logger_level)
+    pha = ui.get_data(idval)
+    assert isinstance(pha, DataPHA)
 
-    def validate_pha(self, idval):
-        """Check that the PHA dataset in id=idval is
-        as expected.
-        """
+    arf = ui.get_arf(idval)
+    assert isinstance(arf, ARF1D)
 
-        self.assertEqual(ui.list_data_ids(), [idval])
+    rmf = ui.get_rmf(idval)
+    assert isinstance(rmf, RMF1D)
 
-        pha = ui.get_data(idval)
-        self.assertIsInstance(pha, DataPHA)
+    bpha = ui.get_bkg(idval, bkg_id=1)
+    assert isinstance(bpha, DataPHA)
 
-        arf = ui.get_arf(idval)
-        self.assertIsInstance(arf, ARF1D)
+    barf = ui.get_arf(idval, bkg_id=1)
+    assert isinstance(barf, ARF1D)
 
-        rmf = ui.get_rmf(idval)
-        self.assertIsInstance(rmf, RMF1D)
+    brmf = ui.get_rmf(idval, bkg_id=1)
+    assert isinstance(brmf, RMF1D)
 
-        bpha = ui.get_bkg(idval, bkg_id=1)
-        self.assertIsInstance(bpha, DataPHA)
+    # normally the background data set would have a different name,
+    # but this is a  PHA Type 3 file.
+    # assert pha.name == bpha.name
+    assert arf.name == barf.name
+    assert rmf.name == brmf.name
 
-        barf = ui.get_arf(idval, bkg_id=1)
-        self.assertIsInstance(barf, ARF1D)
 
-        brmf = ui.get_rmf(idval, bkg_id=1)
-        self.assertIsInstance(brmf, RMF1D)
-
-        # normally the background data set would have a different name,
-        # but this is a  PHA Type 3 file.
-        # self.assertEqual(pha.name, bpha.name)
-        self.assertEqual(arf.name, barf.name)
-        self.assertEqual(rmf.name, brmf.name)
+@requires_fits
+@requires_data
+def test_pha3_read_explicit(make_data_path, clean_astro_ui):
+    """Include .gz in the file name"""
 
-    def testReadExplicit(self):
-        """Include .gz in the file name"""
+    fname = make_data_path(FILE_NAME + '.gz')
+    idval = 12
+    ui.load_pha(idval, fname)
 
-        idval = 12
-        fname = self.head + '_pha3.fits.gz'
-        ui.load_pha(idval, fname)
+    validate_pha(idval)
 
-        self.validate_pha(idval)
+    # TODO: does this indicate that the file name, as read in,
+    #       should have the .gz added to it to match the data
+    #       read in, or left as is?
+    pha = ui.get_data(idval)
+    bpha = ui.get_bkg(idval, bkg_id=1)
+    assert pha.name == bpha.name + '.gz'
+    assert pha.name == fname
```

Comment:
Is the assertion in 87 necessary after doing the validate_pha() check in 87?

Reply:
I haven't looked at this code for a while, but I don't believe `validate_pha` explicitly checks the `name` attribute of the data object, so I believe it is technically testing something "new".